source/framework/vx_image.c

```diff
@@ -376,6 +383,310 @@ static vx_status ownAllocImageBuffer(vx_reference ref)
+        {
+            /* Neither image may be a sub-image */
+            status = (vx_status)VX_ERROR_INVALID_PARAMETERS;
+        }
+        else
+        {
+            tivx_obj_desc_image_t *out_objd = (tivx_obj_desc_image_t *)output->base.obj_desc;
+            vx_uint32 i;
+            vx_bool has_sub_objects = (vx_bool)vx_false_e;
+            for (i = 0; i < TIVX_IMAGE_MAX_SUBIMAGES; ++i)
+            {
+                if ((NULL != input->subimages[i]) ||
+                    (NULL != output->subimages[i]))
+                {
+                    has_sub_objects = (vx_bool)vx_true_e;
+                    break;
+                }
+            }
+            if ((vx_bool)vx_true_e == has_sub_objects)
+            {
+                for (i = 0; i < in_objd->planes; ++i)
+                {
+                    if ((in_objd->mem_size[i] != out_objd->mem_size[i]))
+                    {
+                        VX_PRINT(VX_ZONE_ERROR, "Swapping images with sub-objects and differing allocated memory size is not supported (size 1 = %zu, size 2 = %zu)\n", in_objd->mem_size[i], out_objd->mem_size[i]);
+                        status = (vx_status)VX_ERROR_NOT_SUPPORTED;
+                        break;
+                    }
+                }
+            }
+        }
+    }
+    return status;
+}
+
+static vx_status copyImage(vx_image input, vx_image output)
+{
+    /* Copy entire image by using memcpy only if memory sizes are equal,
+       otherwise use the high-level functions
+       NOTE that this will need updating if uniform images are
+       implemented in a more efficient manner!
+       The images must be copyable!
+     */
+    vx_uint32 i;
+    tivx_obj_desc_image_t *ip_objd = (tivx_obj_desc_image_t *)input->base.obj_desc;
+    tivx_obj_desc_image_t *op_objd = (tivx_obj_desc_image_t *)output->base.obj_desc;
+    vx_status status = (vx_status)VX_SUCCESS;
+    vx_rectangle_t rect = {.start_x = 0, .start_y = 0, .end_x = ip_objd->width, .end_y = ip_objd->height};
+    vx_imagepatch_addressing_t addr;
+    vx_map_id map_id;
+    void *ptr;
+    for (i = 0; i < ip_objd->planes; ++i)
+    {
+        if (ip_objd->mem_size[i] == op_objd->mem_size[i])
+        {
+            tivxCheckStatus(&status, tivxMemBufferMap((void *)(uintptr_t)ip_objd->mem_ptr[i].host_ptr, ip_objd->mem_size[i], 
+                                                      (vx_enum)VX_MEMORY_TYPE_HOST, (vx_enum)VX_READ_ONLY));
+            if ((vx_status)VX_SUCCESS == status)
+            {
+                tivxCheckStatus(&status, tivxMemBufferMap((void *)(uintptr_t)op_objd->mem_ptr[i].host_ptr, ip_objd->mem_size[i], 
+                                                          (vx_enum)VX_MEMORY_TYPE_HOST, (vx_enum)VX_WRITE_ONLY));
+                if ((vx_status)VX_SUCCESS == status)
+                {
+                    (void)memcpy((void *)(uintptr_t)op_objd->mem_ptr[i].host_ptr, (void *)(uintptr_t)ip_objd->mem_ptr[i].host_ptr, ip_objd->mem_size[i]);
```

Comment:
@raphacan Please add a comment here to clarify that single memcpy would be faster than line by line (vxCopyPatch)

Reply:
done